### DISCUSSION TITLE
ダークモードとライトモードを一時的に切り替える方法

### DISCUSSION THREAD

**Discussion by yoshi389111** [increments/qiita-discussions#439] [要望・アイデア・不具合]
Twitter のマイルストーンで紹介されている[記事](https://qiita.com/ishizakiiii/items/5eff79b59bce74fdca0d)を見たら、残念ながらダークモードでは非常に見にくい感じになっていました。

おそらく、白地を前提に透過画像を使っていて、ダークモードでは黒地に黒の画像になってしまっているのではないかと思います。
（逆に将来的には、ダークモードを使っている人の投稿した記事が、ライトモードで見にくいこともあるかもしれないとも思います）

簡単にダークモード⇔ライトモードに切り替えられるスイッチがあったら便利だなと思いました。
（あるいは、上記のような記事がダークモードでも見やすくなればそれでも問題ないと思いますが）

お手数ですが、ご一考願います。

**Comment by munierujp** [comment]
画像が見づらい問題については、GitHubやQiitaに貼る画像はどんな背景色でも見やすいように背景を透過しないのがベターだと思います。
GitHubでも、READMEに貼られた画像の背景が透過されているがゆえにダークモード時に見づらいケースがまれにあります。

https://github.com/standard/standard/issues/1641

とはいえ、デフォルトではシステムの外観設定に合ったテーマで表示しつつ、ユーザーが明示的にテーマを選択できるようにするのは、ユーザーに対して選択肢を与えるという意味では、良いアイデアでしょう。

MDN Web Docs:

<img width="147" alt="スクリーンショット 2023-02-22 0 08 39" src="https://user-images.githubusercontent.com/20086673/220382627-58e1e9e2-5029-45ee-a4d8-d7eae48619e8.png">


YouTube:

<img width="303" alt="スクリーンショット 2023-02-22 0 07 27" src="https://user-images.githubusercontent.com/20086673/220382402-adc8e0ae-8512-4e9f-bde6-fea5e1224810.png">

**Reply by yoshi389111** [reply]
基本的に同意見です。

> 背景を透過しないのがベターだと思います。

そうだと思うのですが（意図して透過させたいケースって思いつかないけど何かあるのかな？）、新規作成分はともかく、過去分をすべて処理するのは大変かもな、と思っています。
また、外部の画像ファイルを埋め込んでいるような場合にも困りますよね。

> デフォルトではシステムの外観設定に合ったテーマで表示しつつ、ユーザーが明示的にテーマを選択できるようにするのは、ユーザーに対して選択肢を与えるという意味では、良いアイデアでしょう。

きっとこれは（何も言わなくても）そのうち実装してくれるんじゃないかと思っているのですが、その時に settings の中とかだと切り替えが大変なので、できれば記事を読んでいるときにすぐに切り替えられる位置にあるといいなぁ、というのが希望です。

**Comment by tanaka-asami415** [comment]
@yoshi389111 @munierujp 
ありがとうございます！🙏
切り替えについては、他にも要望をいただいているため検討します！

**Reply by yoshi389111** [reply]
切り替え機能というタイトルで書いていましたが、透過イメージがダークモードで見難いことがある対策としては、ダークモードでも画像の背景をライトモードと同じにしたらよいような気がしてきましたが、対応は難しいですかね？

（試しに Chrome のデベロッパーツールで `.it-MdContent img` あたりに `background-color` を設定してみたらボーダーがアルファ値を持っていて、透過イメージでは気にならなかったけど、非透過イメージの場合にはボーダーが目立ってしまいました。ボーダーが非透過なら、背景色を変えれば見た目はかなり改善する気がするのですが……）

**Reply by yoshi389111** [reply]
ベータ版で透過画像の対応がされているのを確認しました。

切り替え機能自体は別途検討を進めているとのことで、本件は以上でクローズします。

ありがとうございました。

**Comment by tanaka-asami415** [comment]
@yoshi389111 
以前クローズいただきましたが、切替機能も正式版でリリースできたためご報告させていただきます！✨
詳細は下記ブログをご参照ください🙇
https://blog.qiita.com/release-theme/

リクエストいただきありがとうございました！
